core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinOpExec.scala

```diff
@@ -28,6 +28,30 @@ class HashJoinOpExec[K](
   var currentEntry: Iterator[Tuple] = _
   var currentTuple: Tuple = _
 
+  def getBuildHashTable(): ArrayBuffer[mutable.HashMap[K, ArrayBuffer[Tuple]]] = {
+    val sendingMap = new ArrayBuffer[mutable.HashMap[K, ArrayBuffer[Tuple]]]
+    var count = 1
+    var curr = new mutable.HashMap[K, ArrayBuffer[Tuple]]
+    for ((key, tuples) <- buildTableHashMap) {
+      curr.put(key, tuples._1)
+      if (count % 4000 == 0) {
+        sendingMap.append(curr)
+        curr = new mutable.HashMap[K, ArrayBuffer[Tuple]]
+      }
+      count += 1
+    }
+    if (!curr.isEmpty) sendingMap.append(curr)
+    sendingMap
+  }
+
+  def mergeIntoHashTable(additionalTable: mutable.HashMap[Any, ArrayBuffer[Tuple]]): Unit = {
+    for ((key, tuples) <- additionalTable) {
+      val (storedTuples, _) =
+        buildTableHashMap.getOrElseUpdate(key.asInstanceOf[K], (new ArrayBuffer[Tuple](), false))
+      storedTuples.appendAll(tuples)
```

Comment:
this merge implementation has the assumption that the tuples are unique and no duplicates. you append to the `storedTuples` without deduplicating. is my understanding correct? if so can we document it?

naturally, when talking about "merging" something, the biggest thing I am concerned is about how to handle the conflict/duplicated fields and attributes, etc. I do not see such a case being handled. we either can push back to say this is our assumption, or this method should be renamed to `appendIntoHashTable`.

Reply:
The HashJoin appends whatever is given to it. It assumes that you are not giving it duplicates. Will add comments.